pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/Label.java

```diff
@@ -71,12 +73,15 @@ public String getDisplayName() {
             return "Run on an agent matching a label";
         }
 
+        public AutoCompletionCandidates doAutoCompleteLabel(@QueryParameter String value) {
+            return LabelExpression.autoComplete(value);
+        }
+
         public FormValidation doCheckLabel(@QueryParameter String label) {
-            if (StringUtils.isEmpty(Util.fixEmptyAndTrim(label))) {
+            if (Util.fixEmptyAndTrim(label) == null) {
                 return FormValidation.error("Label is required.");
```

Comment:
Seems to contradict https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/9f9982419a20aa302e60d658908e0283632b1500/pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/Label.java#L50. There does seem to be a potentially valid reason to use a null label instead of just `Any`: specifying `retries`. (But specifying neither seems wrong.)